src/mlpack/core/cv/k_fold_cv.hpp

```diff
@@ -0,0 +1,219 @@
+/**
+ * @file k_fold_cv.hpp
+ * @author Kirill Mishchenko
+ *
+ * k-fold cross-validation.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_K_FOLD_CV_HPP
+#define MLPACK_CORE_CV_K_FOLD_CV_HPP
+
+#include <mlpack/core/cv/meta_info_extractor.hpp>
+#include <mlpack/core/cv/cv_base.hpp>
+
+namespace mlpack {
+namespace cv {
+
+/**
+ * The class KFoldCV implements k-fold cross-validation for regression and
+ * classification algorithms.
+ *
+ * To construct a KFoldCV object you need to pass the k parameter and arguments
+ * that specify data. For the latter see the CVBase constructors as a reference
+ * - the CVBase constructors take exactly the same arguments as ones that are
+ * supposed to be passed after the k parameter in the KFoldCV constructor.
```

Comment:
Correct me if I have misunderstood---I think that because the inheritance is `private`, the constructors of `CVBase` are not available in `KFoldCV`, and instead the single constructor of `KFoldCV` is delegating to `CVBase`.  So in the end, when a user looks at the `KFoldCV` documentation, they only see this overload, and have to dig further to find the `CVBase` constructors that are being referred to.  What do you think, do you like the idea of adding more constructors for clarity, or adding more documentation to be fully clear about what should be passed in?